tests/contracts/test_deposit.py

```diff
@@ -1,22 +1,61 @@
 import pytest
-from conftest import REGISTRATION_DEPOSIT
 
 
 @pytest.mark.parametrize(
     'success,amount_deposit',
     [
-        (True, REGISTRATION_DEPOSIT),
-        (False, REGISTRATION_DEPOSIT - 1),
-        (False, REGISTRATION_DEPOSIT + 1)
+        (True, 32),
+        (False, 31),
+        (False, 33),
+        (False, 0)
     ]
 )
-def test_deposit(tester, a0, w3, success, amount_deposit, assert_tx_failed):
-    base_tester, registration = tester
+def test_deposit(registration_contract, a0, w3, success, amount_deposit, assert_tx_failed):
 
-    call = registration.functions.deposit(b'\x00'*32, 43, a0, b'\x00'*32)
+    call = registration_contract.functions.deposit(
+        b'\x00'*32,
+        43,
+        a0,
+        b'\x00'*32)
     if success:
         assert call.transact({"value": w3.toWei(amount_deposit, "ether")})
     else:
         assert_tx_failed(
             lambda: call.transact({"value": w3.toWei(amount_deposit, "ether")})
         )
+
+
+def test_no_reuse_of_pubkey(registration_contract, a0, w3, assert_tx_failed):
+
+    call = registration_contract.functions.deposit(
+        b'\x00'*32,
```

Comment:
Looks like you're testing 0x00 as the pubkey, not `a0`, right?

Reply:
Oops you're right. Sorry that was an embarrassing mistake 😆